src/Adapter.php

```diff
@@ -4,27 +4,31 @@
 
 namespace Yiisoft\Queue\AMQP;
 
+use BackedEnum;
 use PhpAmqpLib\Message\AMQPMessage;
 use Throwable;
 use Yiisoft\Queue\Adapter\AdapterInterface;
 use Yiisoft\Queue\AMQP\Exception\NotImplementedException;
 use Yiisoft\Queue\Cli\LoopInterface;
 use Yiisoft\Queue\Enum\JobStatus;
+use Yiisoft\Queue\Message\IdEnvelope;
 use Yiisoft\Queue\Message\MessageInterface;
+use Yiisoft\Queue\Message\MessageSerializerInterface;
 
 final class Adapter implements AdapterInterface
 {
     public function __construct(
         private QueueProviderInterface $queueProvider,
-        private MessageSerializerInterface $serializer,
-        private LoopInterface $loop,
+        private readonly MessageSerializerInterface $serializer,
+        private readonly LoopInterface $loop,
     ) {
     }
 
-    public function withChannel(string $channel): self
+    public function withChannel(BackedEnum|string $channel): self
     {
         $instance = clone $this;
-        $instance->queueProvider = $this->queueProvider->withChannelName($channel);
+        $channelName = is_string($channel) ? $channel : (string) $channel->value;
+        $instance->queueProvider = $this->queueProvider->withChannelName($channelName);
```

Comment:
Better add enumeration support to `QueueProviderInterface`

Reply:
Let's create another PR for this